Jetsurvey/app/src/main/java/com/example/compose/jetsurvey/survey/question/SliderQuestion.kt

```diff
@@ -50,60 +64,130 @@ fun SliderQuestion(
     var sliderPosition by remember {
         mutableStateOf(answer?.answerValue ?: possibleAnswer.defaultValue)
     }
-    Column(modifier = modifier) {
-        Row {
-            Slider(
-                value = sliderPosition,
-                onValueChange = {
-                    sliderPosition = it
-                    onAnswerSelected(it)
-                },
-                valueRange = possibleAnswer.range,
-                steps = possibleAnswer.steps,
-                modifier = Modifier
-                    .padding(horizontal = 16.dp)
-                    .fillMaxWidth()
-            )
+    Column(modifier = modifier.fillMaxSize()) {
+        DiscreteSlider(
+            value = sliderPosition.toInt(),
+            onValueChange = { new ->
+                sliderPosition = new.toFloat()
+                onAnswerSelected(new.toFloat())
+            },
+            startText = stringResource(possibleAnswer.startText),
+            neutralText = stringResource(possibleAnswer.neutralText),
+            endText = stringResource(possibleAnswer.endText)
+        )
+    }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+private fun DiscreteSlider(
+    value: Int,
+    onValueChange: (Int) -> Unit,
+    startText: String,
+    neutralText: String,
+    endText: String,
+    modifier: Modifier = Modifier,
+    ticks: Int = 5,
+    onValueChangeFinished: (() -> Unit)? = null,
+    interactionSource: MutableInteractionSource = remember { MutableInteractionSource() }
+) {
+    require(ticks >= 3) { "ticks should be >= 3" }
+    Column {
+        Slider(
+            value = value.toFloat(),
+            onValueChange = { onValueChange(it.roundToInt()) },
+            modifier = modifier.padding(horizontal = 21.dp),
+            steps = ticks - 2,
+            valueRange = 1f..(ticks.toFloat()),
+            onValueChangeFinished = onValueChangeFinished,
+            interactionSource = interactionSource,
+            track = { sliderPositions -> Track(sliderPositions) }
+        )
+
+        Row(
+            horizontalArrangement = Arrangement.SpaceBetween,
+            modifier = Modifier.fillMaxWidth()
+        ) {
+            ProvideTextStyle(
+                MaterialTheme.typography.labelMedium.copy(textAlign = TextAlign.Center)
+            ) {
+                Text(
+                    text = startText,
+                    modifier = Modifier.widthIn(max = 60.dp)
```

Comment:
I think this should be a percentage of the width, ie number of ticks, divided by overall size